controllers/cluster_controller.go

```diff
@@ -918,13 +918,51 @@ func (instance FdbInstance) GetPublicIPs() []string {
 
 	source := instance.Metadata.Annotations[fdbtypes.PublicIPSourceAnnotation]
 	if source == "" || source == string(fdbtypes.PublicIPSourcePod) {
-		// TODO for dual-stack support return PodIPs
-		return []string{instance.Pod.Status.PodIP}
+		return getPublicIpsForPod(instance.Pod)
 	}
 
 	return []string{instance.Pod.ObjectMeta.Annotations[fdbtypes.PublicIPAnnotation]}
 }
 
+func getPublicIpsForPod(pod *corev1.Pod) []string {
+	var podIPPattern *regexp.Regexp
+
+	if pod == nil {
+		return []string{}
+	}
+
+	for _, container := range pod.Spec.Containers {
+		if container.Name == "foundationdb-kubernetes-sidecar" {
+			for indexOfArgument, argument := range container.Args {
+				if argument == "--public-ip-pattern" && indexOfArgument < len(container.Args)-1 {
+					patternString := container.Args[indexOfArgument+1]
+					pattern, err := regexp.Compile(patternString)
+					if err != nil {
+						log.Error(err, "Error parsing public IP pattern", "pattern", pattern)
+						return nil
+					}
+					podIPPattern = pattern
+					break
+				}
+			}
+		}
+	}
+
+	if podIPPattern != nil {
+		podIPs := pod.Status.PodIPs
+		matchingIPs := make([]string, 0, len(podIPs))
+
+		for _, ip := range podIPs {
+			if podIPPattern.Match([]byte(ip.IP)) {
+				matchingIPs = append(matchingIPs, ip.IP)
+			}
+		}
```

Comment:
Wouldn't be easier to just use IPv4 or IPv6 instead of that IP pattern matching?